spec/lib/tasks/update_subscriptions_rake_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe 'subscription_records:convert_subscriptions', type: :task do
+  let!(:old_style_subscription) do
+    create(:subscription, frequency: :daily,
+      search_criteria: { subject: 'english', working_pattern: 'full_time' }.to_json
+    )
+  end
+
+  let!(:new_style_subscription) do
+    create(:subscription, frequency: :daily,
+      search_criteria: { subject: 'english', working_patterns: ['full_time', 'part_time'] }.to_json
+    )
+  end
+
+  let!(:no_working_pattern_subscription) do
+    create(:subscription, frequency: :daily, search_criteria: { subject: 'english' }.to_json)
+  end
+
+  let!(:nil_criteria_subscription) do
+    create(:subscription, frequency: :daily, search_criteria: nil)
+  end
+
+  before do
+    UpdateSubscriptionsWithNewWorkingPatterns.run!
+  end
+
+  it 'Converts a string working pattern to an array working pattern in search criteria' do
```

Comment:
Not a problem here, but you might want to have a look at the `Rspec` `change` pattern matcher: 

https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher

It's ideal for these kinds of specs if you're writing more of them in the future :)